internal/build/job/manager.go

```diff
@@ -138,7 +138,7 @@ func (jbm *jobManager) Sync(
 	case job.Status.Active == 1:
 		return build.Result{Status: build.StatusInProgress, Requeue: true}, nil
 	case job.Status.Failed == 1:
-		return build.Result{}, fmt.Errorf("job failed: %v", err)
+		return build.Result{Status: build.StatusFailed}, fmt.Errorf("job failed: %v", err)
```

Comment:
I don't think we should do this. If the returned error is not nil, no one should use the first returned value.
I think we should rather create an error type with the desired error message, return that, and use `if errors.Is()` in the reconciler.

Reply:
fixed